core/shared/src/main/scala/fs2/internal/Resource.scala

```diff
@@ -147,8 +147,10 @@ private[internal] object Resource {
                   else {
                     // scope is closed and this is last lease, assure finalizer is removed from the state and run
                     F.flatten(state.modify { s =>
-                      // previous finalizer shall be alwayy present at this point, this shall invoke it
-                      s.copy(finalizer = None) -> s.finalizer.getOrElse(F.pure(Right(())))
+                      // previous finalizer shall be always present at this point, this shall invoke it
+                      s.copy(finalizer = None) -> s.finalizer
+                        .map(_(ExitCase.Canceled))
```

Comment:
I think this should be `ExitCase.Completed` as releasing a lease seems like normal completion of the resource. It is just delayed.

Reply:
Thanks, good catch